packages/material-ui/test/integration/MenuList.test.js

```diff
@@ -72,434 +15,554 @@ describe('<MenuList> integration', () => {
     return;
   }
 
-  after(() => {
+  afterEach(() => {
     cleanup();
   });
 
+  specify('the MenuItems have the `menuitem` role', () => {
+    const { getAllByRole } = render(
+      <MenuList>
+        <MenuItem>Menu Item 1</MenuItem>
+        <MenuItem>Menu Item 2</MenuItem>
+        <MenuItem>Menu Item 3</MenuItem>
+      </MenuList>,
+    );
+
+    expect(getAllByRole('menuitem')).to.have.length(3);
+  });
+
   describe('keyboard controls and tabIndex manipulation', () => {
-    let wrapper;
-    let getCommitCount;
+    specify('the specified item should be in tab order while the rest is focusable', () => {
+      const { getAllByRole } = render(
+        <MenuList>
+          <MenuItem tabIndex={0}>Menu Item 1</MenuItem>
+          <MenuItem>Menu Item 2</MenuItem>
+          <MenuItem>Menu Item 3</MenuItem>
+        </MenuList>,
+      );
+
+      expect(getAllByRole('menuitem')[0]).to.have.property('tabIndex', 0);
+      expect(getAllByRole('menuitem')[1]).to.have.property('tabIndex', -1);
+      expect(getAllByRole('menuitem')[2]).to.have.property('tabIndex', -1);
```

Comment:
I have some "managed-focus" matcher in mind but want to collect more use cases before abstracting it away.